### PR TITLE
App Insights Construction and availability as well as login UI enhancements

### PR DESCRIPTION
**Pull request by kpqdap**
The below implements additional app insights fixes as well and the variables at build time.
Currently the Create Tenant on failed login does not work as expected and must be debugged.

### CODE REVIEW

**Comment by rshokqdap** [comment]
Transcribe and App Insights are working after passing the arguments during the docker build.

https://dis-qgcdg.atlassian.net/browse/QGGPT-376

**Review by QDAP-Fred — CHANGES_REQUESTED** [review]

File: src/next.config.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ const securityHeaders = [
 const nextConfig = {
   output: "standalone",
   compiler: {
+    styledComponents: true,
```
Comment (QDAP-Fred) [inline src/next.config.js:63]: this can be removed since we aren't using styled-components

File: src/features/auth/auth-api.ts [hunk 2]
```diff
@@ -100,6 +116,17 @@ export const options: NextAuthOptions = {
...
+    error: "/login-error",
+  },
+  theme: {
+    colorScheme: "dark",
+    brandColor: "#09549f",
+    logo: "/ai-icon.png",
+    buttonText: "Single sign-on in with your Queensland Government Account",
+  },
+  useSecureCookies: true,
+  debug: process.env.NODE_ENV === "development" ? true : false,
```
Comment (QDAP-Fred) [inline src/features/auth/auth-api.ts:129]: can be simplified to `debug: process.env.NODE_ENV === "development",`

File: azure-pipelines.yml [hunk 3]
```diff
@@ -105,10 +105,19 @@ stages:
...
               buildContext: "src"
-              arguments: --build-arg BUILD_ID=$(Build.BuildId)
+              arguments: --build-arg BUILD_ID=$(Build.BuildId) --build-arg APPLICATION_INSIGHTS=$(NEXT_PUBLIC_APPLICATIONINSIGHTS_CONNECTION_STRING) --build-arg FEATURE_TRANSCRIBE_TENANTS=$(NEXT_PUBLIC_FEATURE_TRANSCRIBE_TENANTS) --build-arg FEATURE_TRANSLATOR=$(NEXT_PUBLIC_FEATURE_TRANSLATOR) --build-arg PUBLIC_API_URL=$(NEXT_PUBLIC_API_URL)
               repository: "$(REPOSITORY)"
               tags: "$(Build.BuildId)"
 
+          - task: AzurePowerShell@5
+            displayName: "Prune agent docker images"
+            inputs:
+              azureSubscription: "$(Dev_Service_Connection)"
```
Comment (QDAP-Fred) [inline azure-pipelines.yml:115]: make sure `Dev_Service_Connection` is added in the variable groups

File: src/features/auth/sign-in.ts [hunk 4]
```diff
@@ -4,15 +4,30 @@ import { hashValue } from "./helpers"
 import { User } from "next-auth"
 import { AdapterUser } from "next-auth/adapters"
 
+export enum ErrorType {
```
Comment (QDAP-Fred) [inline src/features/auth/sign-in.ts:7]: I would recommend something like this to be more explicit (name and type wise):
```
export enum SignInErrorType {
  NotAuthorised = "notAuthorised",
  SignInFailed = "signInFailed",
}
type SignInSuccess = {
  success: true
}
type SignInError = {
  success: false
  errorCode: SignInErrorType
}
export type SignInResponse = SignInSuccess | SignInError
```

that way it wouldn't be possible to return a response such as 
```
{
  success: true,
  errorCode: ErrorType.NotAuthorised,
}
```

File: src/features/auth/auth-api.ts [hunk 5]
```diff
@@ -69,13 +72,26 @@ export const options: NextAuthOptions = {
...
+        return false
+      }
+
       try {
         const groups = user?.secGroups ?? []
-        return await UserSignInHandler.handleSignIn(user, groups)
-      } catch (error) {
-        console.error("Error in signIn callback", error)
+        const signInCallbackResponse = await UserSignInHandler.handleSignIn(user, groups)
+        if (signInCallbackResponse.success === true) {
```
Comment (QDAP-Fred) [inline src/features/auth/auth-api.ts:83]: could be simplified:
`if (signInCallbackResponse.success) ...`

**Review by QDAP-Fred — CHANGES_REQUESTED** [review]

**Review by rshokqdap — APPROVED** [review]

**Pull request merged**